### PR TITLE
(update) URL Structure for Checkout and Multi Currency Support (Part 1/2)

### PR DESCRIPTION
**Pull request by thisisayush**
References: #126

Updates the URL structure to match Woocommerce

[woocommerce-plugin/php/WC_Gateway_Blockonomics.php](https://github.com/blockonomics/woocommerce-plugin/blob/master/php/WC_Gateway_Blockonomics.php#L101-L133) -> [modules/gateways/blockonomics/payment.php](https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/bch-url-updates/modules/gateways/blockonomics/payment.php#L24-L62)
[php/Blockonomics.php](https://github.com/blockonomics/woocommerce-plugin/blob/master/php/Blockonomics.php#L366-L375) -> [modules/gateways/blockonomics.php](https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/bch-url-updates/modules/gateways/blockonomics.php#L244-L258)

Update app.js to support new URL structure:
- Add selected crypto
- Removes Crypto Selector from JS App to Crypto Selector Template
- Refactor 'blockonomics_currency' -> 'crypto' in `get_order`

### CODE REVIEW

File: modules/gateways/blockonomics/payment.php [hunk 1]
```diff
@@ -24,54 +24,47 @@
...
+    $active_currencies = $blockonomics->getActiveCurrencies();
+    $ca->assign('active_currencies', $active_currencies);
+    $ca->assign('order_hash', $select_crypto);
+    $ca->setTemplate('/modules/gateways/blockonomics/assets/templates/crypto_options.tpl');
+}else if ($finish_order) {
+    $finish_url = App::getSystemURL() . 'viewinvoice.php?id=' . $finish_order . '&paymentsuccess=true';
+    header("Location: $finish_url");
+    exit();
+}else if ($get_order && $crypto) {
+    $existing_order = $blockonomics->processOrderHash($get_order, $crypto);
```
Comment (DarrenWestwood) [inline modules/gateways/blockonomics/payment.php:55]: Lets refactor this to match woocommerce by moving functionality into the Blockonomics class:
```
if($crypto === "empty"){
    $blockonomics->load_blockonomics_template('no_crypto_selected');
}else if ($show_order && $crypto) {
    $blockonomics->load_checkout_template($show_order, $crypto);
}else if ($select_crypto) {
    $blockonomics->load_blockonomics_template('crypto_options');
}else if ($finish_order) {
    $blockonomics->redirect_finish_order($finish_order);
}else if ($get_order && $crypto) {
    $existing_order = $blockonomics->processOrderHash($get_order, $crypto);
```

**Review by DarrenWestwood — COMMENTED** [review]
The new URL structure is working as desired

**Comment by thisisayush** [comment]
@DarrenWestwood 

New Refactored Changes:
- Add load_blockonomics_template, redirect_finish_order, and load_checkout_template to Blockonomics Class and refactor the code in payment.php
- Add ger_order_checkout_params to Blockonomics Class and refactor the code in gateways/blockonomics.php
- Remove unused Display Error String from Translation
- Remove Commented code from checkout.tpl

File: modules/gateways/blockonomics/assets/templates/no_crypto_selected.tpl [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<link rel="stylesheet" type="text/css" href="{$WEB_ROOT}/modules/gateways/blockonomics/assets/css/order.css">
+
+<div class="bnomics-order-container">
+    <h3>
+        No crypto currencies are enabled for checkout
```
Comment (DarrenWestwood) [inline modules/gateways/blockonomics/assets/templates/no_crypto_selected.tpl:5]: We do not [validate the currencies in woocommerce](https://github.com/blockonomics/woocommerce-plugin/blob/master/blockonomics-woocommerce.php#L244-L254). Lets remove the currency validation from WHMCS as we now have the no_crypto_selected template as with Woocommerce: https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/master/modules/gateways/blockonomics.php#L358-L374

Comment (DarrenWestwood) [inline modules/gateways/blockonomics/assets/templates/no_crypto_selected.tpl:5]: We do not [validate the currencies in woocommerce](https://github.com/blockonomics/woocommerce-plugin/blob/master/blockonomics-woocommerce.php#L244-L254). Lets remove the currency validation from WHMCS as we now have the no_crypto_selected template as with Woocommerce: https://github.com/blockonomics/whmcs-bitcoin-plugin/blob/master/modules/gateways/blockonomics.php#L358-L374

**Comment by shivaenigma** [comment]
Please use translable strings in *modules/gateways/blockonomics/assets/templates/no_crypto_selected.tpl*. Any reason we are not using strings from langs/english.php?

**Comment by thisisayush** [comment]
> Please use translable strings in _modules/gateways/blockonomics/assets/templates/no_crypto_selected.tpl_. Any reason we are not using strings from langs/english.php?

Updated

**Pull request merged**